base/base/ReplxxLineReader.cpp

```diff
@@ -293,116 +456,64 @@ void ReplxxLineReader::addToHistory(const String & line)
+    TemporaryFile editor_file("clickhouse_client_editor_XXXXXX.sql");
+    editor_file.write(rx.get_state().text());
+    editor_file.close();
 
-    static void * real_vfork = dlsym(RTLD_DEFAULT, "vfork");
-    if (!real_vfork)
-    {
-        rx.print("Cannot find symbol vfork in myself: %s\n", errnoToString().c_str());
-        return -1;
-    }
-
-    pid_t pid = reinterpret_cast<pid_t (*)()>(real_vfork)();
-
-    if (-1 == pid)
+    char * const argv[] = {editor.data(), editor_file.getPath().data(), nullptr};
+    try
     {
-        rx.print("Cannot vfork: %s\n", errnoToString().c_str());
-        return -1;
+        if (executeCommand(argv) == 0)
+        {
+            const std::string & new_query = read_file(editor_file.getPath());
+            rx.set_state(replxx::Replxx::State(new_query.c_str(), new_query.size()));
+        }
     }
-
-    /// Child
-    if (0 == pid)
+    catch (const std::runtime_error & e)
     {
-        sigset_t mask;
-        sigemptyset(&mask);
-        sigprocmask(0, nullptr, &mask); // NOLINT(concurrency-mt-unsafe) // ok in newly created process
-        sigprocmask(SIG_UNBLOCK, &mask, nullptr); // NOLINT(concurrency-mt-unsafe) // ok in newly created process
-
-        execvp(editor.c_str(), argv);
-        rx.print("Cannot execute %s: %s\n", editor.c_str(), errnoToString().c_str());
-        _exit(-1);
+        rx.print(e.what());
     }
 
-    int status = 0;
-    do
-    {
-        int exited_pid = waitpid(pid, &status, 0);
-        if (exited_pid == -1)
-        {
-            if (errno == EINTR)
-                continue;
-
-            rx.print("Cannot waitpid: %s\n", errnoToString().c_str());
-            return -1;
-        }
-        else
-            break;
-    } while (true);
-    return status;
+    if (bracketed_paste_enabled)
+        enableBracketedPaste();
 }
 
-void ReplxxLineReader::openEditor()
+void ReplxxLineReader::openInteractiveHistorySearch()
 {
-    char filename[] = "clickhouse_replxx_XXXXXX.sql";
-    int fd = ::mkstemps(filename, 4);
-    if (-1 == fd)
+    TemporaryFile history_file("clickhouse_client_editor_XXXXXX.sql");
+    auto hs(rx.history_scan());
+    while (hs.next())
     {
-        rx.print("Cannot create temporary file to edit query: %s\n", errnoToString().c_str());
-        return;
+        history_file.write(hs.get().text());
+        history_file.write(std::string(1, '\0'));
     }
+    history_file.close();
 
-    replxx::Replxx::State state(rx.get_state());
+    TemporaryFile output_file("clickhouse_client_history_out_XXXXXX.sql");
+    output_file.close();
 
-    size_t bytes_written = 0;
-    const char * begin = state.text();
-    size_t offset = strlen(state.text());
-    while (bytes_written != offset)
+    char sh[] = "sh";
+    char sh_c[] = "-c";
+    std::string fuzzy_finder_command = fmt::format("{} --read0 --height=30% < {} > {}", fuzzy_finder, history_file.getPath(), output_file.getPath());
```

Comment:
maybe use ` --tiebreak=index` to have the results more consistent?

Reply:
Actually `--no-sort --tac` should be added, it is way more convenient, will do.

Reply:
what does `--tac` do?

I just think it would be nice to get the behavior used for history where the latest is shown first and while searching the results just collapse and don't change on each keypress.

Reply:
```
+s, --no-sort
Do not sort the result
[--tac](https://manpages.debian.org/unstable/fzf/fzf.1.en.html#tac)
Reverse the order of the input
e.g.
history | fzf --tac --no-sort
```

great, they actually use this combination for history example